R/board_rsconnect.R

```diff
@@ -549,20 +550,40 @@ rsc_user_name <- function(board, guid) {
 }
 
 read_cache <- function(path) {
-  if (file.exists(path)) {
+  if (file.exists(path) && use_cache()) {
     yaml::read_yaml(path, eval.expr = FALSE)
   } else {
     list()
   }
 }
-update_cache <- function(path, key, value) {
-  cache <- read_cache(path)
-  cache[[key]] <- value
-  write_yaml(cache, path)
 
+update_cache <- function(path, key, value) {
+  if (use_cache()) {
+    cache <- read_cache(path)
+    cache[[key]] <- value
+    write_yaml(cache, path)
+  }
   value
 }
 
+clear_cache <- function(board,
+                        cache = c("content-cache.yml", "users-cache.yml")) {
+  path <- fs::path(board$cache, cache)
+  cli::cli_inform("Deleting {.file {path}}")
+  fs::file_delete(path)
+}
```

Comment:
I added this new little function to delete the cache files. Right now, I have not exported or documented it, which makes it like a helper function for us to use in development and sort of secondarily, something for us to tell people to use when they get stuck. People would use it like:

```r
pins:::clear_cache(b, "content-cache.yml")
```
Thoughts? Should this be a regular exported function instead?

Reply:
If we do really end removing these caches, then this function would need to be deprecated (if it were exported).

Reply:
I decided to export and document it, and we can deal with that later.